File: src/main/python/user_knot.py
from services import UserService as USER_SERVICE


class UserKnot(object):

    def __init__(self, user_id):
        self.user_id = user_id
        self._cached_user = None

    @property
    def user(self):
        if self._cached_user is None:
            self._cached_user = USER_SERVICE.get_user(self.user_id)
        return self._cached_user

File: src/main/java/org/kambanaria/knots/examples/UserKnot.java
package org.kambanaria.knots.examples;

import org.kambanaria.knots.microservices_models.User;
import static org.kambanaria.knots.services.StaticRegistry.USER_SERVICE;

public class UserKnot {

    private final int userId;
    private User user;

    public UserKnot(int userId) {
        this.userId = userId;
    }

    public int getUserId() {
        return userId;
    }

    public User getUser() {
        if (user == null) {
            user = USER_SERVICE.getUserById(userId);
        }
        return user;
    }
}
